### DISCUSSION TITLE
Self-Hosting a headless node to access the GRPC API

### DISCUSSION THREAD

**Discussion by ebanDev** [anyproto/contributors#210] [Self-hosting]
Hello!

I want to have a node that allows me to access the GRPC API, I don’t want to completely self host any-sync I just want to add a node to my current anytype devices, how can I do that?

**Comment by hellodword** [comment]
You can access the gRPC API of anytype-heart (which means the clients) easily: https://github.com/orgs/anyproto/discussions/211

But the `any-sync*` use QUIC/ymux as the transport: https://github.com/anyproto/any-sync/blob/main/net/transport

And they use drpc actually.

**Comment by ebanDev** [comment]
I already played a bit with this API from my computer, but I want to be able to communicate with it even when I computer is closed, so I would like kind of a headless instance of anytype running on a server I own so that this API is always available. For example i'd like to be able to add things to my anytype by let's say using a voice assistant, I would need to be able to communicate with the gRPC API even when my computer isn't powered-on, so having a client in my anytype "network" (I mean my clients in my space, so for now just my phone and my computer and the backup node from AnyType) that is 24/7 avaiable would be really important

**Reply by hellodword** [reply]
Actually it's possible, because the anytype-ts is using this way.

Try:

`env ANYTYPE_GRPC_ADDR=127.0.0.1:8888 ANYTYPE_GRPCWEB_ADDR=127.0.0.1:8889 anytypeHelper`

The `anytypeHelper` is https://github.com/anyproto/anytype-heart/tree/main/cmd/grpcserver

**Reply by ebanDev** [reply]
Amazing this works ! But I have a problem, when running this command I want to create get an app key (using AccountLocalLinkNewChallenge) but there is not pop-up showing the code for the auth challenge as the software is running in CLI-only mode, how could I manage to authenticate ?

**Reply by hellodword** [reply]
I think there're two methods:

1. Get the app key from your headful device, then copy the `~/.config/anytype` directory to your headless device
2. Use the `WalletRecover` and `WalletCreateSession` with the `mnemonic` not with the `appKey`, see https://github.com/anyproto/anytype-heart/blob/a8b159830941fb29437e154eb08dae745c530276/pb/protos/commands.proto#L589-L592
